core/src/main/scala/com/github/johnynek/paiges/Doc.scala

```diff
@@ -246,6 +252,31 @@ object Doc {
   def str[T](t: T): Doc =
     text(t.toString)
 
+  /**
+   * A Doc is empty if and only if all renderings are empty
+   */
+  def isEmpty(d: Doc): Boolean = {
```

Comment:
It might be worth making this truly tail-recursive, using a `stack: List[Doc]` in cases where we needed to branch. What do you think?